tasks/internal/get-advisory-severity/get-advisory-severity.yaml

```
@@ -119,6 +120,100 @@ spec:
               fi
           }
 
+          # Function to process a batch of CVEs with its own token
+          process_cve_batch() { # Expected arguments are [batch_id, batch_cves]
+              echo "Batch $1: Getting token..."
+              set +x
+              batch_token=$(curl --retry 3 --negotiate -u : "${OSIDB_URL}"/auth/token | jq -r '.access')
+              set -x
+
+              echo "Batch $1: Processing CVEs..."
+
+              while read -r cve; do
+                  if [[ -n "$cve" ]]; then
+                      cache_file="$CVE_CACHE_DIR/${cve}.json"
+
+                      # Note: If we bypass this check in a race condition, the output from OSIDB will be
+                      # the same anyways so it is safe - multiple batches writing identical data is harmless
+                      if [[ -f "$cache_file" ]]; then
+                          continue  # Already processed
+                      fi
+
+                      echo "Batch $1: Processing CVE ${cve}"
+                      curl_url="${OSIDB_URL}/osidb/api/v1/flaws?cve_id=${cve}&include_fields=${INCLUDE_FIELDS}"
+                      set +x
+
+                      # Try API call with current token, refresh and retry if it fails
+                      if ! curl --retry 3 \
+                          -H 'Content-Type: application/json' \
+                          -H "Authorization: Bearer ${batch_token}" \
+                          "$curl_url" | jq '.results[0]' > "$cache_file"; then
+                          echo "Batch $1: API call failed, refreshing token and retrying..."
+                          batch_token=$(curl --retry 3 --negotiate -u : "${OSIDB_URL}"/auth/token | jq -r '.access')
+                          curl --retry 3 -H 'Content-Type: application/json' -H "Authorization: Bearer ${batch_token}" \
+                              "$curl_url" | jq '.results[0]' > "$cache_file"
+                      fi
+                      set -x
+                  fi
+              done <<< "$2"
+
+              echo "Batch $1: Completed"
+          }
+
+          ALL_CVES=$(jq -r '[.[] | select(.cves.fixed) | .cves.fixed | keys[]] | unique | .[]' <<< "${IMAGES}" \
+            | tr "\n" " ")
```

Up to you: no error handling for CVE extraction
```suggestion
          if ! ALL_CVES=$(jq -r '[.[] | select(.cves.fixed) | .cves.fixed | keys[]] | unique | .[]' <<< "${IMAGES}" | tr "\n" " "); then
    echo "ERROR: Failed to extract CVEs from releaseNotesImages data" | tee "$STDERR_FILE"
    exit 1
fi
```

---

As talked about in the meeting, let's just trust that our schema validation in check-data-keys and set-advisory-severity worked well so we don't have to validate the CVE structure here

---

sure